packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.tsx

```diff
@@ -0,0 +1,580 @@
+import * as React from 'react';
+import {
+  autobind,
+  BaseComponent,
+  css,
+  findIndex,
+  getId
+} from '../../Utilities';
+import {
+  ISwatchColorPickerProps,
+  ISwatchColorPickerItemProps,
+  SwatchColorPickerItemType,
+  CellShape
+} from './SwatchColorPicker.Props';
+import { DirectionalHint } from '../../ContextualMenu';
+import { getColorFromString } from '../../utilities/color/colors';
+import { Grid } from '../../utilities/Grid/Grid';
+import { DefaultButton, CommandButton } from '../../Button';
+import { Callout } from '../../Callout';
+import { FocusZone } from '../../FocusZone';
+import * as stylesImport from './SwatchColorPicker.scss';
+const styles: any = stylesImport;
+
+export interface ISwatchColorPickerState {
+  selectedIndex?: number;
+  expanded?: boolean;
+}
+
+export class SwatchColorPicker extends BaseComponent<ISwatchColorPickerProps, ISwatchColorPickerState> {
+
+  public static defaultProps = {
+    cellShape: CellShape.circle,
+    updateButtonIconWithColor: false,
+    selectedId: null,
+    disabled: false
+  };
+
+  private _id: string;
+
+  private _buttonWrapper: HTMLDivElement;
+
+  private _numOfItemsInChunk;
+
+  constructor(props: ISwatchColorPickerProps) {
+    super(props);
+
+    this._id = props.id || getId('swatchColorPicker');
+    this._numOfItemsInChunk = -1;
+
+    this.state = {
+      selectedIndex: props.selectedId && this._getSelectedIndex(props.swatchColorPickerItems, props.selectedId),
+      expanded: false
+    };
+  }
+
+  public componentWillReceiveProps(newProps: ISwatchColorPickerProps) {
+    let newSelectedIndex = newProps.selectedId && this._getSelectedIndex(newProps.swatchColorPickerItems, newProps.selectedId);
+
+    if ((newSelectedIndex !== undefined && newSelectedIndex !== null) &&
+      (newSelectedIndex !== this.state.selectedIndex || newProps.swatchColorPickerItems !== this.props.swatchColorPickerItems)) {
```

Comment:
The call to getSelectedIndx is at the top of this method, isn't the check for newSelectedIndex sufficient? The fact that swatchColorPickerItems changed doesn't seem to matter here. Am I missing something?